dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java

```diff
@@ -439,7 +442,58 @@ private boolean fireUpdateIdentifiers(String oldVal, String newVal) {
   public List<ContentType> findUrlMapped() throws DotDataException {
     return fac.findUrlMapped();
   }
+  
+  @Override
+  public ContentType save(ContentType contentType, List<Field> fields) throws DotDataException, DotSecurityException {
+	  return save(contentType, fields, new ArrayList<FieldVariable>());
+  }
+
+  @Override
+  public ContentType save(ContentType contentType, List<Field> fields, List<FieldVariable> fieldVariables) throws DotDataException, DotSecurityException {
+	  ContentType localContentType = save(contentType);
+	  
+	  //Fields on Receiver
+      List<Field> localFields = localContentType.fields();
+      //Create a copy in order to avoid a possible concurrent modification error
+      List<String> localFieldsVarNames = new ArrayList<String>();
+
+      for (Field localField : localFields) {
+      	localFieldsVarNames.add(localField.variable());
+      }
 
+      //for each field in the pushed structure lets create it if doesn't exists and update its properties if it does
+      for( Field field : fields ) {
+    	  try {
+	    	  Field localField = fAPI.byContentTypeIdAndVar(field.contentTypeId(), field.variable());
+	    	  if ( localField == null || !UtilMethods.isSet( localField.inode() ) ) {
```

Comment:
Since we are already catching the Exception Thrown when the field doesn't exist and save it, this *if* is not doing anything.